### PR TITLE
Do not create default broker client SA and its RBAC

### PR DESCRIPTION
**Pull request by Jaanki**
When deploying broker, `submariner-k8s-broker-client` SA is created and bound to `submariner-k8s-broker-cluster`. This SA is not used anywhere.

This PR removes this SA.

### CODE REVIEW

File: pkg/broker/rbac.go [hunk 1]
```diff
@@ -29,7 +29,6 @@ import (
 const (
 	submarinerBrokerClusterRole      = "submariner-k8s-broker-cluster"
 	submarinerBrokerAdminRole        = "submariner-k8s-broker-admin"
```
Comment (skitt) [inline pkg/broker/rbac.go]:
```suggestion
	submarinerBrokerClusterRole = "submariner-k8s-broker-cluster"
	submarinerBrokerAdminRole   = "submariner-k8s-broker-admin"
```

**Review by skitt — CHANGES_REQUESTED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Comment by dfarrell07** [comment]
Should this be backported @Jaanki?

**Comment by tpantelis** [comment]
This PR is causing a [system test failure](https://github.com/submariner-io/submariner-operator/actions/runs/4729438042/jobs/8402138154?pr=2601#step:3:1309) in **submariner-operator**. It uses the `submariner-k8s-broker-client` SA/secret to obtain [SUBMARINER_BROKER_CA](https://github.com/submariner-io/submariner-operator/blob/7134634809a8b0407e9ae6198d5fe331c3009a05/scripts/test/system.sh#L16). It's also referenced in [submariner-charts](https://github.com/submariner-io/submariner-charts/blob/61b7d37f76cf19bc81043e0c77fc734696ff244a/submariner-k8s-broker/templates/NOTES.txt#L9). Perhaps we were premature in removing it or the uses/references elsewhere need to removed/updated? @Jaanki @skitt

**Comment by skitt** [comment]
Let’s revert it for now.

**Comment by skitt** [comment]
We should have other SAs that can be used for those purposes, let’s update the test and chart docs to use one of those.

**Comment by Jaanki** [comment]
> Should this be backported @Jaanki?

Not needed @dfarrell07. This is a part of cleanup/sycn I am doing between subctl and operator to use same RBAC definitions defined in operator.

**Comment by Jaanki** [comment]
> We should have other SAs that can be used for those purposes, let’s update the test and chart docs to use one of those.

operator PR: https://github.com/submariner-io/submariner-operator/pull/2602

**Pull request merged**